sql/src/test/java/io/crate/integrationtests/AnyIntegrationTest.java

```diff
@@ -126,4 +126,20 @@ public void testArrayReferenceOfDifferentTypeSoThatCastIsRequired() throws Excep
         execute("refresh table t");
         execute("select * from t where 4 < ANY (x) ");
     }
+
+    @Test
+    public void testNotAnyInWhereClause() {
```

Comment:
Please make the name more concrete - that this shouldn't filter out empty arrays should imho be made explicit.
Maybe it would also be possible to rewrite this as a LuceneQueryBuilder unit test